src/librustc_driver/driver.rs

```diff
@@ -243,7 +248,48 @@ pub fn compile_input(sess: &Session,
                 tcx.print_debug_stats();
             }
 
-            let trans = phase_4_translate_to_llvm(tcx, analysis, incremental_hashes_map,
+            #[cfg(not(feature="llvm"))]
+            {
```

Comment:
I'm also not sure about the idea of creating dylibs that are actually rlibs.

Reply:
Without llvm we cant create real dylibs, but many rustc crates are compiled to dylibs, so by emitting rlibs instead we can compile all rustc libs.